packages/cloud-cognitive/src/components/SidePanel/SidePanel.js

```diff
@@ -577,11 +447,16 @@ SidePanel.propTypes = {
    * Sets the primary action buttons for the side panel
    */
   primaryActions: PropTypes.oneOfType([
-    validateActions(),
+    ActionSet.validateActions(
```

Comment:
Should we add all of the possible sizes of the side panel here? Looks like `large` and `max` are missing

Reply:
I didn't want to have the ActionSet explicitly know about extraSmall/small/medium/large/max, as these apply to the SidePanel but might not apply in other places where the ActionSet is useful. So instead I abstracted the decisions that were being made into two settings:
* When validating the actions, you were applying tighter constraints for extraSmall/small/medium sizes (limit of three buttons, no mixing of ghost buttons), so I gave the validator a parameter which selects whether to use the 'small' constraints or the normal constraints. We use the 'small' constraints when props.size is extraSmall/small/medium. I've added a comment to try and make that rather mysterious line a bit clearer!
* When you're going to stack the buttons, you reverse the order of them, so I gave ActionSet a 'reverse' prop.
These mean that ActionSet doesn't know specifically about when to switch the order or when to apply the tighter constraints -- that's SidePanel's decision.

Reply:
On further examination of the Tearsheet and the styling, I think it will be better to replace the `reverse` prop and the small parameter with actually having `size` on the ActionSet, so the SidePanel just passes its own size through. That also enables ActionSet to take the scss from current SidePanel. I will update this PR when I've done that.